### ISSUE TITLE
Error when updating aws_lb_listener_rule

### ISSUE THREAD

**Issue by tutida** [terraform-providers/terraform-provider-aws#11339]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.12
+ provider.aws v2.42.0
+ provider.template v2.1.2
```

### Affected Resource(s)

* aws_lb_listener_rule

### Terraform Configuration Files

```hcl
resource "aws_lb_listener_rule" "http-listener-rule" {
  listener_arn = "${aws_alb_listener.http-listener.arn}"
  priority     = 100

  action {
    type             = "forward"
    target_group_arn = "${aws_alb_target_group.target_group_http.arn}"
  }

  condition {
    http_request_method {
      values = ["GET"]
    }
  }

  condition {
    source_ip {
      values = ["192.168.0.0/16"]
    }
  }
}
```

### Expected Behavior

ListerRule has both `http_request_method` and` source_ip` conditions.

When `source_ip` is applied after applying` http_request_method `

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_lb_listener_rule.http-listener-rule-get will be updated in-place
  ~ resource "aws_lb_listener_rule" "http-listener-rule" {
        arn          = "arn:aws:earn:aws:xxxxx
        id           = "arn:aws:arn:aws:xxxxx"
        listener_arn = "arn:aws:arn:aws:xxxxx
        priority     = 100

        action {
            order            = 1
            target_group_arn = "arn:aws:arn:aws:xxxxx
            type             = "forward"
        }

        condition {
            field  = "http-request-method"
            values = []

            http_request_method {
                values = [
                    "GET",
                ]
            }
        }
      + condition {
          + field  = (known after apply)
          + values = (known after apply)

          + host_header {
              + values = (known after apply)
            }

          + path_pattern {
              + values = (known after apply)
            }

          + source_ip {
              + values = [
                  + "192.168.0.0/16",
                ]
            }
        }
    }

```

### Actual Behavior

An error occurs:
```
Error: Both field and values must be set in a condition block

  on aws_alb.tf line 42, in resource "aws_lb_listener_rule" "http-listener-rule-get":
  42: resource "aws_lb_listener_rule" "http-listener-rule-get" {
```

**Comment by dpiddockcmp** [comment]
This should have been fixed in v2.43.0. Please try with the new version.

**Comment by bflad** [comment]
I'm going to optimistically mark this as closed with the fix merged in version 2.43.0 released yesterday, but please reach out if upgrading does not fix this. 👍
